### PR TITLE
DS-4380:  Update XOAI (OAI-PMH) for full JDK11 compatibility

### PR DESCRIPTION
**Pull request by tdonohue**
Minor PR to update to XOAI v3.3.0, which is more compatible with JDK11.  In XOAI v3.3.0, I've updated dependencies to avoid any "An illegal reflective access operation has occurred" warnings when running JDK 9 or above.  See https://github.com/DSpace/xoai/releases/tag/xoai-3.3.0

This PR also makes a minor refactor to the classes which obtain the Solr configuration for `dspace-oai` to ensure they provide better logging & use the more correct version of the configuration `oai.solr.url`. 

I've tested both of these minor changes locally and they work well on latest `master` and with JDK11.

### CODE REVIEW

**Review by abollini — APPROVED** [review]
code looks correct and changes trivial... so I'm ok in trusting travis if it will report success

**Comment by tdonohue** [comment]
At +1 and Travis approved.  Merging, as this is a tiny PR & I've tested it thoroughly locally.

File: dspace-oai/src/main/java/org/dspace/xoai/solr/DSpaceSolrServer.java [hunk 1]
```diff
@@ -30,12 +30,12 @@ private DSpaceSolrServer() { }
 
     public static SolrClient getServer() throws SolrServerException {
         if (_server == null) {
+            String serverUrl = ConfigurationManager.getProperty("oai.solr.url");
```
Comment (mwoodiupui) [inline dspace-oai/src/main/java/org/dspace/xoai/solr/DSpaceSolrServer.java]: Switch to ConfigurationService?

File: dspace-oai/src/main/java/org/dspace/xoai/solr/DSpaceSolrServer.java [hunk 2]
```diff
@@ -30,12 +30,12 @@ private DSpaceSolrServer() { }
...
             try {
-                _server = new HttpSolrClient.Builder(
-                    ConfigurationManager.getProperty("oai", "solr.url")).build();
-                log.debug("Solr Server Initialized");
+                _server = new HttpSolrClient.Builder(serverUrl).build();
+                log.debug("OAI Solr Server Initialized");
             } catch (Exception e) {
-                log.error(e.getMessage(), e);
+                log.error("Could not initialize OAI Solr Server at " + serverUrl , e);
```
Comment (mwoodiupui) [inline dspace-oai/src/main/java/org/dspace/xoai/solr/DSpaceSolrServer.java]: Parameterize?

File: dspace-oai/src/main/java/org/dspace/xoai/services/impl/solr/DSpaceSolrServerResolver.java [hunk 3]
```diff
@@ -26,11 +26,12 @@
...
         if (server == null) {
+            String serverUrl = configurationService.getProperty("oai.solr.url");
             try {
-                server = new HttpSolrClient.Builder(configurationService.getProperty("oai", "solr.url")).build();
-                log.debug("Solr Server Initialized");
+                server = new HttpSolrClient.Builder(serverUrl).build();
+                log.debug("OAI Solr Server Initialized");
             } catch (Exception e) {
-                log.error(e.getMessage(), e);
+                log.error("Could not initialize OAI Solr Server at " + serverUrl , e);
```
Comment (mwoodiupui) [inline dspace-oai/src/main/java/org/dspace/xoai/services/impl/solr/DSpaceSolrServerResolver.java]: Message could be parameterized:  `log.error("Could not initializa OAI Solr Server at {}", serverUrl, e);`
Unlikely to save many cycles, but it's a good habit to get into.

**Review by mwoodiupui — COMMENTED** [review]
+1 by inspection.  This makes sense to me.  I noted a couple of minor tweaks which might be made.

**Pull request merged**